src/engine/runtime.js

```diff
@@ -321,6 +386,160 @@ class Runtime extends EventEmitter {
     }
 
     /**
+     * Register the primitives provided by an extension.
+     * @param {ExtensionInfo} extensionInfo - information about the extension (id, blocks, etc.)
+     * @private
+     */
+    _registerExtensionPrimitives (extensionInfo) {
+        const categoryInfo = {
+            id: extensionInfo.id,
+            name: extensionInfo.name,
+            color1: '#FF6680',
```

Comment:
If these color definitions are now going to come in through the VM should we find a good place to define / manage them globally?

Reply:
Yep - ideally I'd like these to somehow come from the GUI but I haven't yet come up with a good way to do that -- actually I haven't put a whole lot of thought into it yet. Input welcome!